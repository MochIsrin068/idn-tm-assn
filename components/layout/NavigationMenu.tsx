import React from "react";
import { message } from "antd";
import { ArrowLeftOutlined, ShareAltOutlined } from "@ant-design/icons";
import Link from "next/link";

type TPropsNavigationMenu = {
  children?: JSX.Element;
};

export default function NavigationMenu({ children }: TPropsNavigationMenu) {
  return (
    <div className="flex items-center justify-between">
      <Link href="/">
        <ArrowLeftOutlined className="cursor-pointer text-xl font-bold" />
      </Link>
      {children}
      <ShareAltOutlined
        className="cursor-pointer text-xl font-bold"
        onClick={() => message.success("Share this content!")}
      />
    </div>
  );
}
