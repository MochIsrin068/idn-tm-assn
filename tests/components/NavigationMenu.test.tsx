import React from "react";
import { render, screen } from "@testing-library/react";
import NavigationMenu from "../../components/layout/NavigationMenu";

describe("tests components NavigationMenu", () => {
  it("Perform snapshots test NavigationMenu", () => {
    const inputDefaultTree = render(
      <NavigationMenu>
        <div>IDN Media - News</div>
      </NavigationMenu>
    );
    expect(inputDefaultTree).toMatchSnapshot();
  });

  it("test text if have a childrenn", () => {
    render(
      <NavigationMenu>
        <div>IDN Media - News</div>
      </NavigationMenu>
    );
    const label = screen.getByText(/IDN Media - News/i);
    expect(label).toBeInTheDocument();
  });

  it("test should not have text of children", () => {
    render(<NavigationMenu />);
    const label = screen.queryByText(/IDN Media - News/i);
    expect(label).not.toBeInTheDocument();
  });
});
