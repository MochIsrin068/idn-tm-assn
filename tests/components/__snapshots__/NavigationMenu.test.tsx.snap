// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests components NavigationMenu Perform snapshots test NavigationMenu 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-between"
      >
        <span
          aria-label="arrow-left"
          class="anticon anticon-arrow-left cursor-pointer text-xl font-bold"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="arrow-left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
        <div>
          IDN Media - News
        </div>
        <span
          aria-label="share-alt"
          class="anticon anticon-share-alt cursor-pointer text-xl font-bold"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="share-alt"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 000-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
            />
          </svg>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-between"
    >
      <span
        aria-label="arrow-left"
        class="anticon anticon-arrow-left cursor-pointer text-xl font-bold"
        role="img"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          data-icon="arrow-left"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
      <div>
        IDN Media - News
      </div>
      <span
        aria-label="share-alt"
        class="anticon anticon-share-alt cursor-pointer text-xl font-bold"
        role="img"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          data-icon="share-alt"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 000-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
          />
        </svg>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
